src/hooks/sarcoToken/useApprove.ts

```diff
@@ -0,0 +1,14 @@
+import { ethers } from 'ethers';
+import { useContractWrite } from 'wagmi';
+import { SarcoTokenABI } from '../../abis/SarcoToken';
+
+export function useApprove() {
+  const { write } = useContractWrite({
```

Comment:
We are not getting the toast or transaction stuff from useSubmitTransaction(). so maybe we need to rethink that a little. Although. to the best of my knowledge, this is the only contract write that's not through the diamond deployer.

Reply:
Yeah... This also needs to use the `useSubmitTransaction` hook.

Reply:
Actually this is one case where we are not using the diamond contract address so it can be treated separately from the other transaction calls.